packages/grafana-ui/src/components/Card/Card.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { cloneElement, FC, HTMLAttributes, ReactElement, ReactNode } from 'react';
+import { css, cx } from 'emotion';
+import { GrafanaTheme } from '@grafana/data';
+import { useTheme, styleMixins } from '../../themes';
+import { Tooltip, PopoverContent } from '../Tooltip/Tooltip';
+import { OnTagClick } from '../Tags/Tag';
+import { TagList } from '../Tags/TagList';
+
+interface ContainerProps extends HTMLAttributes<HTMLOrSVGElement> {
+  /** Customise the container html element for the card. Defaults to div */
+  tag?: keyof JSX.IntrinsicElements;
+  /** Content for the card's tooltip */
+  tooltip?: PopoverContent;
+}
+
+const CardContainer: FC<ContainerProps> = ({ tag = 'div', children, tooltip, ...props }) => {
+  const element = React.createElement(tag, props, children);
+  return tooltip ? (
+    <Tooltip placement="top" content={tooltip} theme="info">
+      {element}
+    </Tooltip>
+  ) : (
+    element
+  );
+};
+
+export interface CardInnerProps {
+  href?: string;
+}
+
+const CardInner: FC<CardInnerProps> = ({ children, href }) => {
+  const theme = useTheme();
+  const styles = getStyles(theme);
+  return href ? (
+    <a className={styles.innerLink} href={href}>
+      {children}
+    </a>
+  ) : (
+    <>{children}</>
+  );
+};
+
+export interface Props extends ContainerProps {
+  /** Main heading for the Card **/
+  heading: ReactNode;
+  /** Additional data about the card. If array is supplied, elements will be rendered with vertical line separator */
+  metaData?: ReactNode | ReactNode[];
+  /** Card description text */
+  description?: string;
+  /** List of tags to display in the card */
+  tags?: string[];
+  /** Optional callback for tag onclick event */
+  onTagClick?: OnTagClick;
+  /** Indicates if the card and all its actions can be interacted with */
+  disabled?: boolean;
+  /** Image or icon to be displayed on the let side of the card */
+  mediaContent?: ReactNode;
+  /** Main card actions **/
+  actions?: ReactElement[];
+  /** Right-side actions */
+  secondaryActions?: ReactElement[];
+  /** Link to redirect to on card click. If provided, the Card inner content will be rendered inside `a` */
+  href?: string;
+  /** On click handler for the Card */
+  onClick?: () => void;
+}
+
+export const Card: FC<Props> = ({
+  heading,
+  description,
+  metaData,
+  tags = [],
+  onTagClick,
+  disabled,
+  mediaContent,
+  actions = [],
+  tooltip,
+  secondaryActions = [],
+  tag,
+  href,
+  onClick,
+  className,
+  ...htmlProps
+}) => {
+  const hasActions = Boolean(actions.length || secondaryActions.length);
+  const disableHover = disabled || actions.length > 1;
```

Comment:
Is this correct? I see BasicCard story has a pointer cursor, which does not make much sense. I think you should also check for `href` property here.

Reply:
Good catch1 It'd also check if `onClick` is provided.